service/payments/src/main/java/nl/tudelft/wdm/group1/payments/events/UserConsumer.java

```diff
@@ -28,9 +29,10 @@ public void consumePaymentSuccessful(Payment payment) {
     }
 
     @KafkaListener(topics = {UsersTopics.CREDIT_SUBTRACTED_FOR_PAYMENT_FAILED})
-    public void consumePaymentFailed(Payment payment) {
+    public void consumePaymentFailed(Payment payment) throws ResourceNotFoundException {
         logger.info("Consuming [{}] -> {}", UsersTopics.CREDIT_SUBTRACTED_FOR_PAYMENT_FAILED, payment);
         // Emit payment failed message
         producer.emitPaymentFailed(payment);
+        paymentRepository.delete(paymentRepository.findOrElseThrow(payment.getOrderId()));
```

Comment:
Maybe it's better to set a state on the payment?

Also I'm not sure it is certain that the payment actually exists here. Previously there was an error because the payment did not exists at this specific location.

Reply:
Well, the paymentid is linked to an orderId as primary key. The initial idea of deleting a payment is to ensure that an order can be paid later.
I think previous time the error rose due to multiple listeners on the same topic. Not due to deletion of the payment.

Reply:
The tests illustrate as well that the payment can be deleted safely